dbt/include/spark/macros/materializations/incremental/incremental.sql

```diff
@@ -29,7 +29,9 @@
   {% if existing_relation is none %}
     {% set build_sql = create_table_as(False, target_relation, sql) %}
   {% elif existing_relation.is_view or full_refresh_mode %}
-    {% do adapter.drop_relation(existing_relation) %}
+    {% if existing_relation.is_view or file_format != 'delta' %}
+      {% do adapter.drop_relation(existing_relation) %}
+    {% endif %}
```

Comment:
Two quick thoughts here:
- I think this conditional logic can be simplified to avoid the nested `if` statements
- We need to check both that the table _already exists_ in Delta format, and _will be_ created in Delta format

So further up:
```
{% set is_delta = (file_format == 'delta' and existing_relation.is_delta) %}
```
And then:
```sql
  {% elif existing_relation.is_view or (full_refresh_mode and not is_delta) %}
    {% do adapter.drop_relation(existing_relation) %}
```